server/src/main/java/org/opensearch/common/util/CuckooFilter.java

```diff
@@ -541,4 +513,148 @@ public boolean equals(Object other) {
             && Objects.equals(this.count, that.count)
             && Objects.equals(this.evictedFingerprint, that.evictedFingerprint);
     }
+
+    /**
+     * A compact implementation of Lucene's Packed64 class.
+     * Previously, this code was part of `:server module under org.apache.lucene.util.packed.Packed64`.
+     * This shorter version has been added here to resolve the JPMS split package issue.
+     */
+    private static class Packed64 {
```

Comment:
Even `final` could be applied:

```suggestion
    private static final class Packed64 {
```